ocimotel/repoclient.go

```diff
@@ -37,23 +37,30 @@ func NewOciRepo(ref *ociMotelReference, sys *types.SystemContext) (r OciRepo, er
 		port = fmt.Sprintf("%d", ref.port)
 	}
 
-	insecureSkipVerify := (sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue)
-	tlsClientConfig := &tls.Config{
-		InsecureSkipVerify: insecureSkipVerify,
-	}
+	var tlsClientConfig *tls.Config
+	insecureSkipVerify := false
+	creds := ""
+
+	if sys != nil {
```

Comment:
I guess not, since sys.DockerAuthConfig gets dereferenced later.

Could you change this to just a

```
if sys == nil {
    return nil, fmt.Errorf("nil systemcontext")
}
```
or explain why you want to do any of the rest of the function if sys is nil?